src/OrbitGl/GpuTrack.cpp

```diff
@@ -54,312 +46,234 @@ std::string MapGpuTimelineToTrackLabel(std::string_view timeline) {
   }
 
-  return color;
+  marker_track_->SetPos(pos_[0], current_y);
 }
 
-float GpuTrack::GetYFromTimer(const TimerInfo& timer_info) const {
-  auto adjusted_depth = static_cast<float>(timer_info.depth());
+float GpuTrack::GetHeight() const {
   if (collapse_toggle_->IsCollapsed()) {
-    adjusted_depth = 0.f;
+    return submission_track_->GetHeight();
   }
-  if (timer_info.type() == TimerInfo::kGpuDebugMarker) {
-    return pos_[1] - layout_->GetTextBoxHeight() * (adjusted_depth + 1.f);
+  float height = 0.f;
+  if (!submission_track_->IsEmpty()) {
+    height += submission_track_->GetHeight();
   }
-  CHECK(timer_info.type() == TimerInfo::kGpuActivity ||
-        timer_info.type() == TimerInfo::kGpuCommandBuffer);
-
-  // Command buffer timers are drawn underneath the matching "hw execution" timer, which has the
-  // same depth value as the command buffer timer. Therefore, we need to double the depth in the
-  // case that we have command buffer timers.
-  if (has_vulkan_layer_command_buffer_timers_) {
-    adjusted_depth *= 2.f;
+  if (!marker_track_->IsEmpty()) {
+    height += marker_track_->GetHeight();
   }
-
-  float gap_space = adjusted_depth * layout_->GetSpaceBetweenGpuDepths();
-
-  // Command buffer timers have the same depth value as their matching "hw execution" timer.
-  // As we want to draw command buffers underneath the hw execution timers, we need to increase
-  // the depth by one.
-  if (timer_info.type() == TimerInfo::kGpuCommandBuffer) {
-    adjusted_depth += 1.f;
+  if (!submission_track_->IsEmpty() && !marker_track_->IsEmpty()) {
+    height += layout_->GetSpaceBetweenGpuSubtracks();
   }
-  return pos_[1] - layout_->GetTextBoxHeight() * (adjusted_depth + 1.f) - gap_space;
-}
-
-// When track is collapsed, only draw "hardware execution" timers and the root "debug markers".
-bool GpuTrack::TimerFilter(const TimerInfo& timer_info) const {
-  if (collapse_toggle_->IsCollapsed()) {
-    std::string gpu_stage = string_manager_->Get(timer_info.user_data_key()).value_or("");
-    return gpu_stage == kHwExecutionString ||
-           (timer_info.type() == TimerInfo::kGpuDebugMarker && timer_info.depth() == 0);
+  if (!(submission_track_->IsEmpty() && marker_track_->IsEmpty())) {
+    height += layout_->GetSpaceBetweenGpuSubtracks();
   }
-  return true;
+  return height;
 }
 
-void GpuTrack::SetTimesliceText(const TimerInfo& timer_info, double elapsed_us, float min_x,
-                                float z_offset, TextBox* text_box) {
-  if (text_box->GetText().empty()) {
-    std::string time = GetPrettyTime(absl::Microseconds(elapsed_us));
+void GpuTrack::Draw(GlCanvas* canvas, PickingMode picking_mode, float z_offset) {
+  float track_height = GetHeight();
+  float track_width = canvas->GetWorldWidth();
 
-    text_box->SetElapsedTimeTextLength(time.length());
+  SetPos(canvas->GetWorldTopLeftX(), pos_[1]);
+  SetSize(track_width, track_height);
```

Comment:
I wonder if all of this is really needed... You probably just copied this from the other Tracks, but this seems somewhere between unneeded and broken:
* `GetHeight()` is only correct after drawing, so this is incorrect for at least one frame
* `SetPos` doesn't change anything since `canvas->GetWorldTopLeftX()` always returns 0

But if I'm not mistaken all Tracks suffer from this, so let's keep it consistent and clean it up at some point.